cpp/core/shuffle/LocalPartitionWriter.cc

```diff
@@ -420,6 +420,7 @@ arrow::Status LocalPartitionWriter::mergeSpills(uint32_t partitionId) {
   auto spillIter = spills_.begin();
   while (spillIter != spills_.end()) {
     ARROW_ASSIGN_OR_RAISE(auto st, dataFileOs_->Tell());
+    (*spillIter)->openForRead(options_.shuffleFileBufferSize);
```

Comment:
Could you fix the failed UTs? It's likely the failures are caused by this change. You should add `spill->openForRead(options_.shuffleFileBufferSize)` before line 460
https://github.com/apache/incubator-gluten/pull/7861/files#diff-d58dca9ecda139d81d2630e1adb22ae01953ad4de0a9c086bd07c07619a3a927R460